### ISSUE TITLE
Mult-Threading

### ISSUE THREAD

**Issue by jacobfriedman** [anchore/syft#2233]
**What would you like to be added**:
Multi-Threading

**Why is this needed**:
My superior is waiting 20 minutes for this to generate and is telling me I'm taking too much of his time with the tool.

**Additional context**:
Jason "Higher up the ladder" C***:
Tell the syft developers to get their shit together and make it multi-threaded. We've been multi-CPU / mult-core for only a decade or two now...

**Comment by wagoodman** [comment]
This is already supported (see the configuration section in the README.md). The set of catalogers invoked can be run concurrently via configuration:
```
#.syft.yaml

# maximum number of workers used to process the list of package catalogers in parallel
parallelism: 5
```

or via environment variable:
```
SYFT_PARALLELISM=5 syft myimage:latest
```

Please provide more details in another issue about performance issues you are encountering with details about the asset causing the issues to discuss further.

As a reminder for future comments, please keep verbiage appropriate and refrain from derogatory comments.

**Comment by jacobfriedman** [comment]
There's a large system being created in our organization (~80gb) and it took about 20 minutes to generate the SBOM for the image. My apologies for the tone.

Using multiple cores/threads, or, workers, *should* be automatic on this kind of software assuming there's no real trade-off. Something to consider in future.

Search for "RUN syft" language:Dockerfile on github. The time-save with automatic workers will help conserve energy. Think of the 🐧s.
